src/usr/bin/diffuse.py.in

```diff
@@ -2352,6 +2353,239 @@ def _get_rcs_repo(path, prefs):
         # the user specified an invalid folder name
         pass
 
+# SRC support
+class _Src:
+    def __init__(self, root):
+        self.root = root
+        self.url = None
+
+    def _getVcs(self):
+        return 'src'
+
+    def _getURLPrefix(self):
+        return 'URL: '
+
+    def _parseStatusLine(self, s):
+        # src status prints eg "M<TAB>filename" - as opposed to svn which expands the TAB!
+        if len(s) < 3 or s[0] not in 'ACDMR':
+            return
+        return s[0], s[2:]
+
+    def _getPreviousRevision(self, prefs, rev):
+        if rev is None:
+            return 'BASE'
+        m = int(rev)
+        if m > 1:
+            return str(m - 1)
+
+    def _getURL(self, prefs):
+        if self.url is None:
+            vcs, prefix = self._getVcs(), self._getURLPrefix()
+            n = len(prefix)
+            args = [ prefs.getString(vcs + '_bin'), 'info' ]
+            for s in popenReadLines(self.root, args, prefs, vcs + '_bash'):
+                if s.startswith(prefix):
+                    self.url = s[n:]
+                    break
+        return self.url
+
+    def getFileTemplate(self, prefs, name):
+        # FIXME: verify this
+        # merge conflict
+        left = glob.glob(name + '.merge-left.r*')
+        right = glob.glob(name + '.merge-right.r*')
+        if len(left) > 0 and len(right) > 0:
+            return [ (left[-1], None), (name, None), (right[-1], None) ]
+        # update conflict
+        left = sorted(glob.glob(name + '.r*'))
+        right = glob.glob(name + '.mine')
+        right.extend(glob.glob(name + '.working'))
+        if len(left) > 0 and len(right) > 0:
+            return [ (left[-1], None), (name, None), (right[0], None) ]
+        # default case
+        return [ (name, self._getPreviousRevision(prefs, None)), (name, None) ]
+
+    def _getCommitTemplate(self, prefs, rev, names):
+        # there's no concept like a 'commit' for src - each file
+        # has an independent history. So you can either:
+
+        # diffuse -c 14 todo.org
+        # or
+        # diffuse -m [files]
+
+        # Nothing else really makes much sense to me so let's assume that if we have a rev
+        # then we're doing a -m otherwise it's a -c
+
+        # FIXME: removed, added, conflicting files - only modified files work!!
+
+        result = []
+
+        # build command
+        vcs = self._getVcs()
+        vcs_bin, vcs_bash = prefs.getString(vcs + '_bin'), vcs + '_bash'
+        args = [ vcs_bin, 'status' ]
+        if rev is None:
+            if names == [ '.' ]:
+                names = glob.glob("*")
+            prev = 'BASE'
+        else:
+            prev = None
+            args.append(names[0])
+
+        #    args = [ vcs_bin, 'diff', '-c', rev + "-" ]
```

Comment:
Dead code?

If it is, better remove it :wink:

Reply:
No objection to removing that comment. Just cruft.